h2o-core/src/test/java/water/util/MathUtilsTest.java

```diff
@@ -0,0 +1,65 @@
+package water.util;
+
+import junit.framework.TestCase;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static water.util.StringUtils.join;
+
+public class MathUtilsTest extends TestCase {
+
+
+    @Test
+    public void testCompare() throws Exception {
+        assertTrue(MathUtils.compare(1,1, 0.1, 0.1));
+        assertTrue(MathUtils.compare(1,1, 0.0, 0.1));
+        assertTrue(MathUtils.compare(1,1, 0.1, 0.0));
+        assertTrue(MathUtils.compare(1,1, 0.0, 0.0));
+
+        assertTrue(MathUtils.compare(-1,-1, 0.1, 0.1));
+        assertTrue(MathUtils.compare(-1,-1, 0.0, 0.1));
+        assertTrue(MathUtils.compare(-1,-1, 0.1, 0.0));
+        assertTrue(MathUtils.compare(-1,-1, 0.0, 0.0));
+
+        assertTrue(MathUtils.compare(1, 0.900001, 0.1, 0));
+        assertTrue(MathUtils.compare(1, 1.099, 0.1, 0));
+        assertTrue(MathUtils.compare(-1, -0.90001, 0.1, 0));
+        assertTrue(MathUtils.compare(-1, -1.099, 0.1, 0));
+        
+        assertFalse(MathUtils.compare(1, 0.8999999, 0.1, 0));
+        assertFalse(MathUtils.compare(1, 1.10000001, 0.1, 0));
+        assertFalse(MathUtils.compare(-1, -0.899999, 0.1, 0));
+        assertFalse(MathUtils.compare(-1, -1.1000001, 0.1, 0));
+
+        assertTrue(MathUtils.compare(0.99999, 0, 1, 0));
+        assertTrue(MathUtils.compare(-0.99999, 0, 1, 0));
+        assertTrue(MathUtils.compare(0, 0.99999, 1, 0));
+        assertTrue(MathUtils.compare(0, -0.99999, 1, 0)); 
+        
+        assertFalse(MathUtils.compare(1.000001, 0, 1, 0));
+        assertFalse(MathUtils.compare(-1.000001, 0, 1, 0));
+        assertFalse(MathUtils.compare(0, 1.000001, 1, 0));
+        assertFalse(MathUtils.compare(0, -1.000001, 1, 0)); 
+        
+        assertTrue(MathUtils.compare(1, 0.9000, 0.1, 0.1));
+        assertTrue(MathUtils.compare(1, 1.099, 0, 0.1));
+        assertTrue(MathUtils.compare(-1, -0.900, 0.1, 0.1));
+        assertTrue(MathUtils.compare(-1, -1.099, 0, 0.1));
+        
+        assertFalse(MathUtils.compare(1, 0.8999999, 0, 0.1));
+        assertFalse(MathUtils.compare(1,  1.1,0, 0.09));
+        assertFalse(MathUtils.compare(-1, -0.899999, 0, 0.1));
+        assertFalse(MathUtils.compare(-1, -1.100001, 0, 0.1));
+
+        assertFalse(MathUtils.compare(1.0000, 0, 0, 0.1));
```

Comment:
is there a point to all the zeroes after 1.?

Reply:
No, it was just a relic of generating more tests semi-automatically to cover more cases I was thinking about. I have removed it.